zebra-state/src/service/arbitrary.rs

```diff
@@ -86,3 +91,46 @@ impl Strategy for PreparedChain {
         })
     }
 }
+
+/// Generate a chain to test legacy chain check.  
+pub(crate) fn partial_nu5_chain_strategy(
+    transaction_version_override: u32,
+    transaction_has_valid_network_upgrade: bool,
+    blocks_after_nu_activation: u32,
+) -> impl Strategy<
+    Value = (
+        Network,
+        Height,
+        zebra_chain::fmt::SummaryDebug<Vec<Arc<Block>>>,
+    ),
+> {
+    (
+        any::<Network>(),
+        NetworkUpgrade::reduced_branch_id_strategy(),
+    )
+        .prop_flat_map(move |(network, random_nu)| {
+            // We are going to test the legacy chain with Canopy instead as we don't have a
+            // Nu5 activation height yet.
```

Comment:
I understand this can be confusing.

Reply:
Suggestion: We could make it less confusing by passing a network upgrade argument to the strategy.

Reply:
I made this suggestion at https://github.com/ZcashFoundation/zebra/pull/2366/commits/d593c9f009a7843c984aa6ab7f2f2d241e4ba286